server/src/main/java/org/opensearch/index/shard/IndexingStats.java

```diff
@@ -83,6 +197,12 @@ public Stats(StreamInput in) throws IOException {
             noopUpdateCount = in.readVLong();
             isThrottled = in.readBoolean();
             throttleTimeInMillis = in.readLong();
+
+            if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
+                docStatusStats = in.readOptionalWriteable(DocStatusStats::new);
+            } else {
+                docStatusStats = null;
+            }
         }
 
         public Stats(
```

Comment:
why do want to keep this constructor around?

Reply:
Yeah, it's only usage is in `InternalIndexingStats`. We can remove this constructor now. Class is also marked as `@opensearch.internal`.